src/test_websockets.ml

```diff
@@ -0,0 +1,75 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open OUnit
+
+module TestWsIteratee = Websockets.Wsprotocol(Test.StringMonad)
+module I = Iteratees.Iteratee(Test.StringMonad)
+open TestWsIteratee
+open I
+
+let teststr = "\x81\x05\x48\x65\x6c\x6c\x6f"
+
+let get_str str =
+  let ret = ref "" in
+  let i = ref 6 in
+  while !i < String.length str do
+    for j = 0 to 1 do
+      if (!i + j) < String.length str then
+      begin
+        i := !i + j;
+        ret := Printf.sprintf "%s%c" !ret str.[!i]
+      end
+    done;
+    i := !i + 3;
+  done;
+  !ret
+
+let test_wsframe () =
```

Comment:
@jonludlam I have updated this custom test to check frame string, If it looks ok then i can updated other tests too on this format.